client/my-sites/plugins/plugin-list-header/index.jsx

```diff
@@ -0,0 +1,181 @@
+		sites: React.PropTypes.object.isRequired,
+		plugins: React.PropTypes.array.isRequired,
+		selected: React.PropTypes.array.isRequired,
+		isWpCom: React.PropTypes.bool,
+		pluginUpdateCount: React.PropTypes.number
+	},
+
+	onBrowserLinkClick() {
+		analytics.ga.recordEvent( 'Plugins', 'Clicked Add New Plugins' );
+	},
+
+	canAddNewPlugins() {
+		if ( config.isEnabled( 'manage/plugins/browser' ) ) {
+			return this.hasJetpackSelectedSites();
+		}
+		return false;
+	},
+
+	canUpdatePlugins() {
+		return this.props.plugins
+			.filter( plugin => plugin.selected )
+			.some( plugin => plugin.sites.some( site => site.canUpdateFiles ) );
+	},
+
+	hasJetpackSelectedSites() {
+		const selectedSite = this.props.sites.getSelectedSite();
+		if ( selectedSite ) {
+			return !! selectedSite.jetpack;
+		}
+		return this.props.sites.getJetpack().length > 0;
+	},
+
+	unselectOrSelectAll() {
+		if ( this.props.haveSelected ) {
+			PluginsActions.selectPlugins( this.props.sites.getSelectedOrAllWithPlugins(), 'none' );
+			analytics.ga.recordEvent( 'Plugins', 'Clicked to Uncheck All Plugins' );
+			return;
+		}
+		PluginsActions.selectPlugins( this.props.sites.getSelectedOrAllWithPlugins(), 'all' );
+		analytics.ga.recordEvent( 'Plugins', 'Clicked to Check All Plugins' );
+	},
+
+	renderCurrentActionButtons( isWpCom ) {
+		let buttons = [];
+		let rightSideButtons = [];
+		let leftSideButtons = [];
+		let updateButtons = [];
+		let activateButtons = [];
+
+		const hasWpcomPlugins = this.props.selected.some( property( 'wpcom' ) );
+		const isJetpackSelected = this.props.plugins.some( plugin => plugin.selected && 'jetpack' === plugin.slug );
+		const needsRemoveButton = this.props.selected.length && ! hasWpcomPlugins && this.canUpdatePlugins() && ! isJetpackSelected;
+		if ( ! this.props.isBulkManagementActive ) {
+			if ( ! isWpCom && 0 < this.props.pluginUpdateCount ) {
+				rightSideButtons.push(
+					<ButtonGroup key="plugin-list-header__buttons-update-all">
+						<Button compact primary onClick={ this.props.updateAllPlugins } >
+							{ this.translate( 'Update All', { context: 'button label' } ) }
+						</Button>
+					</ButtonGroup>
+				);
+			}
+			rightSideButtons.push(
+				<ButtonGroup key="plugin-list-header__buttons-bulk-management"><Button compact onClick={ this.props.toggleBulkManagement }>{ this.translate( 'Edit All', { context: 'button label' } ) }</Button></ButtonGroup>
+			);
+			if ( ! isWpCom && this.canAddNewPlugins() ) {
+				const selectedSite = this.props.sites.getSelectedSite();
+				const browserUrl = '/plugins/browse' + ( selectedSite ? '/' + selectedSite.slug : '' );
+
+				rightSideButtons.push(
+					<ButtonGroup key="plugin-list-header__buttons-browser"><Button compact href={ browserUrl } onClick={ this.onBrowserLinkClick } className="plugin-list-header__browser-button"><Gridicon key="plus-icon" icon="plus-small" size={ 12 } /><Gridicon key="plugins-icon" icon="plugins" size={ 18 } /></Button></ButtonGroup>
```

Comment:
what do you mean with long? they fit perfectly fine in a single screen!

![image](https://cloud.githubusercontent.com/assets/1554855/11952660/804dcee8-a89a-11e5-9e56-58d686b3490a.png)


:D